src/coreclr/vm/assemblynative.cpp

```diff
@@ -155,15 +153,22 @@ Assembly* AssemblyNative::LoadFromPEImage(AssemblyBinder* pBinder, PEImage *pIma
 
     if (hr != S_OK)
     {
-        // Give a more specific message for the case when we found the assembly with the same name already loaded.
+        StackSString name;
+        spec.GetDisplayName(0, name);
         if (hr == COR_E_FILELOAD)
         {
-            dwMessageID = IDS_HOST_ASSEMBLY_RESOLVER_ASSEMBLY_ALREADY_LOADED_IN_CONTEXT;
+            // Give a more specific message for the case when we found the assembly with the same name already loaded.
+            // Show the assembly name, since we know the error is about the assembly name.
+            StackSString errorString;
+            errorString.LoadResource(CCompRC::Error, IDS_HOST_ASSEMBLY_RESOLVER_ASSEMBLY_ALREADY_LOADED_IN_CONTEXT);
+            COMPlusThrow(kFileLoadException, IDS_EE_FILELOAD_ERROR_GENERIC, name, errorString);
```

Comment:
Nit - not for this PR: I think ideally we would revisit all of the `FileLoadException` construction and change it such that a file path is always included (unless it was loaded from stream, which we should also note). This change improves some of the code paths to use the file path, which is really great. I just think we can do even better.

The reasoning is that in almost all cases, assembly load exceptions happen when loading assemblies from disk. The errors are also almost solely caused by loading the wrong file, either because the process selected the wrong path, or the file on disk in that location is wrong. Debugging these failures basically always starts with trying to figure out the file path - and inspecting that file manually. The exception should include the file path.

I would argue that the exception should even include the path of the already loaded assembly if it plays a role in the failure. So for example MVID mismatch would actually include two paths: 
`Assembly 'test, 1.0.0.0' is already loaded from 'C:\path\test.dll'. Can't load 'C:\path_another\test.dll`.` (or something better).

Or similarly the case where a wrong version is available:
`Can't resolve assembly reference 'test, 2.0.0.0'. Already loaded 'test, 1.0.0.0' from 'C:\path\test.dll'.`

For "stream loads" we could change it to something like:
`Can't resolve assembly reference 'test, 2.0.0.0'. Already loaded 'test, 1.0.0.0' from memory.`

This is based on some of the experiences I had debugging assembly load exceptions - where I had to use the tracing, but in like 50% cases I only needed the file paths involved. The other 50% are related to ALCs, so we may want to revisit that as well and include the ALC in the exception as well (Eventually) :-)

Reply:
I agree we could/should do better. I also think always including both the assembly name and the path (if available) for whatever is being loaded could be useful (rather than choosing one as I did in this PR).

Definitely like the inclusion of whatever assembly it conflicted with. Propagating the details out of whatever level they are currently in is probably going to be the annoying part.